programs/mango-v4/src/state/mango_account_components.rs

```diff
@@ -587,129 +786,145 @@ mod tests {
         assert_eq!(pos.quote_running_native, 400);
         assert_eq!(pos.avg_entry_price(&market), 20.0);
         assert_eq!(pos.break_even_price(&market), 20.0);
-        assert_eq!(pos.realized_pnl_native, I80F48::from(0));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(0));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, 0);
     }
 
     #[test]
     fn test_quote_entry_long_decreasing_from_short() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, -10, 100);
         // Go long 5 @ 50
         pos.record_trade(&mut market, 5, I80F48::from(-250));
         assert_eq!(pos.quote_running_native, -150);
         assert_eq!(pos.avg_entry_price(&market), 10.0); // Entry price remains the same when decreasing
         assert_eq!(pos.break_even_price(&market), -30.0); // Already broke even
-        assert_eq!(pos.realized_pnl_native, I80F48::from(-200));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(-200));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, -5 * 10 / 5 - 1);
     }
 
     #[test]
     fn test_quote_entry_short_decreasing_from_long() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, 10, -100);
         // Go short 5 @ 50
         pos.record_trade(&mut market, -5, I80F48::from(250));
         assert_eq!(pos.quote_running_native, 150);
         assert_eq!(pos.avg_entry_price(&market), 10.0); // Entry price remains the same when decreasing
         assert_eq!(pos.break_even_price(&market), -30.0); // Already broke even
-        assert_eq!(pos.realized_pnl_native, I80F48::from(200));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(200));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, 5 * 10 / 5 + 1);
     }
 
     #[test]
     fn test_quote_entry_long_close_with_short() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, 10, -100);
         // Go short 10 @ 25
         pos.record_trade(&mut market, -10, I80F48::from(250));
         assert_eq!(pos.quote_running_native, 0);
         assert_eq!(pos.avg_entry_price(&market), 0.0); // Entry price zero when no position
         assert_eq!(pos.break_even_price(&market), 0.0);
-        assert_eq!(pos.realized_pnl_native, I80F48::from(150));
+        assert_eq!(pos.realized_trade_pnl_native, I80F48::from(150));
+        assert_eq!(pos.settle_pnl_limit_realized_trade, 10 * 10 / 5 + 1);
     }
 
     #[test]
     fn test_quote_entry_short_close_with_long() {
-        let mut market = PerpMarket::default_for_tests();
+        let mut market = test_perp_market(10.0);
         let mut pos = create_perp_position(&market, -10, 100);
```

Comment:
nitnit: fn create_perp_position would be easier to read if it takes the price per lot instead of quote position native, so something like  `create_perp_position(&market, -10, 10);`